doc/md/generating-ent-schemas.md

```diff
@@ -0,0 +1,216 @@
+---
+id: generating-ent-schemas title: Generating Schemas
+---
+
+## Introduction
+
+:::info This feature is experimental, APIs are subject to change in the future.
+:::
+
+To facilitate the creation of tooling that generates `ent.Schema`s programmatically, `ent` supports the manipulation of
+the `schema/` directory using the `entgo.io/contrib/schemast` package.
+
+## API
+
+### Loading
+
+In order to manipulate an existing schema directory we must first load it into a `schemast.Context` object:
+
+```go
+package main
+
+import (
+	"fmt"
+
+	"entgo.io/contrib/schemast"
+)
+
+func main() {
+	ctx, err := schemast.Load("./ent/schema")
+	if err != nil {
+		panic(err)
+	}
+	if ctx.HasType("user") {
+		fmt.Println("schema directory contains a schema named User!")
+	}
+}
+```
+
+### Printing
+
+To print back out our context to a target directory, use `schemast.Print`:
+
+```go
+package main
+
+import (
+	"fmt"
+
+	"entgo.io/contrib/schemast"
+)
+
+func main() {
+	ctx, err := schemast.Load("./ent/schema")
+	if err != nil {
+		panic(err)
+	}
+	if err := schemast.Print("./ent/schema"); err != nil { // A no-op since we did not manipulate the Context at all.
+		panic(err)
+	}
```

Comment:
```suggestion
	// A no-op since we did not manipulate the schemast.Context at all.
	if err := schemast.Print("./ent/schema"); err != nil {
		panic(err)
	}
```